lib/admin/admin-server.js

```diff
@@ -132,6 +132,44 @@ app.post('/api/machines', (req, res) => {
     .then(() => dbNotify())
 })
 
+/**
+* @api {post} /machines/:machineId/actions/:action
+* @apiName Do :action on specified :machineId
+*
+* @apiParam  {String} [machineId] machine
+* @apiParam  {String} [action] action
+*/
+app.post('/api/machines/:machineId/actions/:action', (req, res, next) => {
+  const actions = ['unpair', 'reboot']
+  const { machineId: machine, action } = req.params
+
+  if (!actions.includes(action)) {
+    return res.status(400).send({
+      statusCode: 400,
+      error: 'Bad Request',
+      message: `:action must be one of ${actions}`
+    })
+  }
+
+  machineLoader.getMachines()
+    .then(machines => machines.find(({ deviceId }) => deviceId === machine))
+    .then(found => found ? next() : res.status(404).send({
+      statusCode: 404,
+      error: 'Not Found',
+      message: `:machine ${machine} does not exists`
+    }))
+})
+
+app.post('/api/machines/:machineId/actions/:action', (req, res) => {
+  const { machineId: deviceId, action } = req.params
+
+  machineLoader.setMachine({ deviceId, action })
+    .then(() => machineLoader.getMachineNames())
+    .then(machines => machines.find(m => m.deviceId === deviceId))
+    .then(machine => res.send(machine))
+    .then(() => dbNotify())
+})
+
```

Comment:
This should be at the new-server, we're running a new front-end and back-end for this stuff.

Reply:
actually this is already on this page, not sure why it's showing as new code on github PR